### ISSUE TITLE
KeyError: 'storageUri' raised around boot_diagnostics and tags

### ISSUE THREAD

**Issue by hemebond** [ansible/ansible_collections_azure#443]
##### SUMMARY
When supplying tags to `azure.azcollection.azure_rm_virtualmachine` a KeyError exception is raised. Similar to #137 

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
azure.azcollection.azure_rm_virtualmachine

##### ANSIBLE VERSION
```
ansible 2.10.6
  config file = ~/.../ansible.cfg
  configured module search path = ['~/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = ~/.local/lib/python3.6/site-packages/ansible
  executable location = ~/.local/bin/ansible
  python version = 3.6.9 (default, Jan 26 2021, 15:33:00) [GCC 8.4.0]
```

##### CONFIGURATION
```
DEFAULT_ASK_PASS = False
DEFAULT_HOST_LIST = [...]
DEFAULT_REMOTE_USER = ...;
HOST_KEY_CHECKING = False
```

##### OS / ENVIRONMENT
Ubuntu Bionic

##### STEPS TO REPRODUCE
Include `boot_diagnostics:enabled:no` and `tags` in the `azure_rm_virtualmachine` task

```yaml
  - name: Create VM
    azure.azcollection.azure_rm_virtualmachine:
      ...
      boot_diagnostics:
        enabled: no
      image:
        offer: UbuntuServer
        publisher: Canonical
        sku: '18.04-LTS'
        version: latest
      tags:
        role: testing
```

##### ACTUAL RESULTS
```The full traceback is:
Traceback (most recent call last):
  File "~/.ansible/tmp/ansible-tmp-1614735033.7236388-3919-185877276540659/AnsiballZ_azure_rm_virtualmachine.py", line 102, in <module>
    _ansiballz_main()
  File "~/.ansible/tmp/ansible-tmp-1614735033.7236388-3919-185877276540659/AnsiballZ_azure_rm_virtualmachine.py", line 94, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "~/.ansible/tmp/ansible-tmp-1614735033.7236388-3919-185877276540659/AnsiballZ_azure_rm_virtualmachine.py", line 40, in invoke_module
    runpy.run_module(mod_name='ansible_collections.azure.azcollection.plugins.modules.azure_rm_virtualmachine', init_globals=None, run_name='__main__', alter_sys=True)
  File "/usr/lib/python3.6/runpy.py", line 205, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_payload_txe9pz1a/ansible_azure.azcollection.azure_rm_virtualmachine_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_virtualmachine.py", line 2215, in <module>
  File "/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_payload_txe9pz1a/ansible_azure.azcollection.azure_rm_virtualmachine_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_virtualmachine.py", line 2211, in main
  File "/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_payload_txe9pz1a/ansible_azure.azcollection.azure_rm_virtualmachine_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_virtualmachine.py", line 894, in __init__
  File "/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_payload_txe9pz1a/ansible_azure.azcollection.azure_rm_virtualmachine_payload.zip/ansible_collections/azure/azcollection/plugins/module_utils/azure_rm_common.py", line 441, in __init__
  File "/tmp/ansible_azure.azcollection.azure_rm_virtualmachine_payload_txe9pz1a/ansible_azure.azcollection.azure_rm_virtualmachine_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_virtualmachine.py", line 1536, in exec_module
KeyError: 'storageUri'
```

**Comment by Fred-sun** [comment]
@hemebond Thank you for raising the issue. We will look into it as soon as possible. Thank you very much!

**Comment by Fred-sun** [comment]
fixes by #565
